Add API for Spaceagon frontboard. (#319)
emfcamp/badge-2024-documentation: docs/tildagon-apps/reference/badge-hardware.md

        # do something
   ```

> There are alternative groups of types, for the 2024 frontboard the alternatives are:

Reviewer: ```suggestion
The EMF 2026 badge uses a new frontboard with a different physical layout from the 2024 badge. It also adds an alternative groups of types. Both groups of types work for the buttons on the 2024 frontboard and the 2026 frontboard.
```

<!-- prettier-ignore -->
| Button | Event Type | Name |
| ------ | ---------- | ---- |
| `"A"` | `BUTTON_TYPES["UP"]` and `FRONTBOARD_BUTTON_TYPES["A"]` | `"A"` |
| `"B"` | `BUTTON_TYPES["RIGHT"]` and `FRONTBOARD_BUTTON_TYPES["B"]` | `"B"` |
| `"C"` | `BUTTON_TYPES["CONFIRM"]` and `FRONTBOARD_BUTTON_TYPES["C"]` | `"C"` |
| `"D"` | `BUTTON_TYPES["DOWN"]` and `FRONTBOARD_BUTTON_TYPES["D"]` | `"D"` |
| `"E"` | `BUTTON_TYPES["LEFT"]` and `FRONTBOARD_BUTTON_TYPES["E"]` | `"E"` |
| `"F"` | `BUTTON_TYPES["CANCEL"]` and `FRONTBOARD_BUTTON_TYPES["F"]` | `"F"` |

> for the 2026 frontboard types see [Spaceagon](tildagon-apps/reference/spaceagon.md/#Spaceagon)

Reviewer: ```suggestion
For more information on the 2026 frontboard types see [Spaceagon](tildagon-apps/reference/spaceagon.md/#Spaceagon).
```

### Methods

> In order to support alternative imu devices a base set of functionality is provided along with i2c access to the device for apps to add functionality. An identify function is provided to determine which device is on the badge. Only the prototype badges have a different imu, but this may change in the future. On boards that have a compass available

Reviewer: Also missing continuation of sentence
```suggestion
In order to support alternative IMU devices a base set of functionality is provided along with I2C access to the device for apps to add functionality. An identify function is provided to determine which device is on the badge. Only the prototype badges have a different imu, IMU this may change in the future. On boards that have a compass available
```

| `id()` | Get the device id | None | `id`: string id of device. |
| `readfrom()` | Read from a device register| `(register address, length)`: address to start read from, length of read | `data`: bytes of the data or -ve error code. |
| `writeto()` | Write data to a device register |  `(register address, bytes)`: address to start write to, data.| `error`: or None if ok. |

> | `mag_read()` | Get the magnetometer data if available | None | `(x,y,z)`: The magnetometer data as a tuple of floats (guass). |

Reviewer: ```suggestion
| `mag_read()` | Get the magnetometer (compass) data if available. Only works on badges with the 2026 frontboard attached. | None | `(x,y,z)`: The magnetometer data as a tuple of floats (guass). |
```
